File: 0205-isomorphic-strings/solution.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        mapping = {}
        for i in range(len(t)):
            if (s[i] not in mapping):
                if t[i] in mapping.values():
                    return False
                mapping[s[i]] = t[i]
            else:
                if (mapping[s[i]] != t[i]):
                    return False
        return True

        

File: 0205-isomorphic-strings/solution.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        char mapS[256] = {0}, mapT[256] = {0};
        for (int i{0}; i < s.length(); i++)
        {
            if (mapS[s[i]] && mapS[s[i]] != t[i]) return false;
            if (mapT[t[i]] && mapT[t[i]] != s[i]) return false;

            mapS[s[i]] = t[i];
            mapT[t[i]] = s[i];
        }
        return true;
    }
};
